language_bindings/rust/src/build.rs

```diff
@@ -0,0 +1,330 @@
+        }
+        //);self
+    }
+    /// Sets the console debug for Halide generators by default this is false
+    pub fn debug(mut self, b:bool) ->Self{
+        self.debug=b;
+        self
+    }
+    /*
+    pub fn make_runtime(self) {
+        if self.generators.is_empty() {
+            //todo make gen-gen
+        }
+        else {
+           let _gen = self.generators[0].make_runtime();
+        }
+    }
+
+     */
+}
+///This is a generator built using [GenBuilder].new_gen()
+///
+impl Generator<'static> {
+    ///Make the generator executable using halide gengen and g++
+    pub fn compile(&self) -> Output {
+
+        let mut cmd_compile = Command::new("g++");
+        cmd_compile.args(["-std=c++17"]);
+
+        // Inlcude flags
+        cmd_compile.args(["-I", self.halide_path.join("include").to_str().unwrap()]);
+        cmd_compile.args(["-I", self.halide_path.join("tools").to_str().unwrap()]);
+
+        // Linker flag
+        cmd_compile.args(["-L", self.halide_path.join("lib").to_str().unwrap()]);
+
+        // Output flag
+        cmd_compile.args(["-o", self.gen_exe.to_str().unwrap()]);
+
+        let temp = self
+            .halide_path
+            .join("tools")
+            .join("GenGen")
+            .with_extension("cpp");
+
+        cmd_compile.args([
+            "-g",
+            self.gen_path.to_str().unwrap(),
+            temp.to_str().unwrap(),
+        ]);
+
+        cmd_compile.args(self.gcc_flags.clone());
+        //compile.args(["-Wl,-rpath","-Wl,/home/rootbutcher2/CLionProjects/Halide-Rusts-tests/Halide/distrib/lib/"]);
+        cmd_compile.output().expect("Make generator failed")
+
+    }
+    /// Runs the previously made executable
+    ///
+    /// Panics if unable to find or run the executable
+    ///```ignore
+    ///     gen.run_gen()
+    ///```
+    pub fn run_gen(&self) -> Output {
+        //assert!(!self.gen_exe.is_none());
+        println!(
+            "cargo:rustc-link-search=native={}",
+            self.rs_output.to_str().unwrap()
+        );
+
+        let mut gen = Command::new(self.gen_exe.to_str().unwrap());
+        gen.args(["-g", self.gen_name.as_str()]);
+        gen.args(["-f", self.gen_name.as_str()]);
+
+        //Todo change to build dir
+        gen.args(["-o", self.rs_output.to_str().unwrap()]);
+        gen.env("LD_LIBRARY_PATH", self.halide_path.join("lib"));
+
+        //Todo add debug
+        if !self.debug {
+            gen.arg(self.target.as_str());
+        } else {
+            let mut temp = String::from(self.target.as_str());
+            temp.push_str("-debug");
+            gen.arg(temp);
+        }
+        gen.output().expect("failed to run")
+    }
+
+    /// Make the Halide runtime
+    ///
+    /// The Halide runtime is required for using halide and contains the buffer_t and other useful functions
+    pub fn make_runtime(&self) -> Result<()> {
+        println!("Cmd to runt: {:?}", self.gen_exe.to_str().unwrap());
```

Comment:
"Command to run" (not "runt")

Reply:
These also kind of look like debug prints. Can we add some `debug() <<` equivalent that also honors the existing environment variables?